avr-gcc-libc: binutils 2.25->2.26, gcc 4.8.4->5.3.0, avr-libc 1.8.1->2.0

ne0phyte opened the pull request:
###### Things done:
- [ ] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
###### More

Updated the packages fetched by avr-gcc-libc to support more recent Atmel MCUs.
I wasn't sure if this should go into a separate nix package (because of the gcc 4.x -> 5.x change).

ne0phyte committed:
avr-gcc-libc: binutils 2.25->2.26,gcc 4.8.4->5.3.0,avr-libc 1.8.1->2.0
All packages updated to support more recent Atmel MCUs.

ne0phyte commented:
The package build is hitting the 4mb travis log limit...
What's best practice in this case? Running make with ---silent?

ne0phyte commented:
I turned off echoing of the gcc and avr-libc build and now the build went through.

fpletz commented:
@ne0phyte Sorry, but please remove the `--silent` again. It's indeed a bit unfortunate that travis doesn't support longer build logs, but our primary CI system hydra does. And we do want to have that kind of verbosity.

Looks good otherwise. Will test this on some firmware later today.

ne0phyte commented:
I removed the `--silent` flags and updated the commit. I didn't test all resulting binaries, but I was able to build a couple of personal projects, including one targeting a previously unsupported Atmel MCU, so it seems to be working fine.

ne0phyte commented:
Any update on this? Would love to see this get merged.
I could also create a separate package for this to prevent breaking things.

bjornfor commented:
Built locally, did small runtime test (avr-gcc produces elf files), no build regressions in dependees. Applied to master (7c6b575c53990234f3ccae580fef13e8ab5180e3). Thanks!
